lib/src/ansi_code.dart

```diff
@@ -0,0 +1,278 @@
+// Copyright 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io' as io;
+
+/// The type of code represented by [AnsiCode].
+class AnsiCodeType {
+  /// The name of the type.
```

Comment:
Remove the name.

I like the `Type` thing because it helps reason about how these things can be combined. Multiple styles: fine. Multiple colors: borked.

etc